### PR TITLE
Webauthn support

### LINKED ISSUE #53: WebAuthn support
- Implement [webauthn support for the sdk](https://github.com/privacyidea/sdk-java/issues/14)
- Add [our lib for webauthn](https://github.com/privacyidea/webauthn-client) and use it/show option to use if a webauthn token was triggered

### LINKED ISSUE #58: TransactionID should be sent to /validate/check for all token challenges
The transactionID should be returned to the /validate/check endpoint in the response to every challenge request, not just the calls for the push token.

This allows support for SMS, Email and other challenge tokens which are triggered by the user providing the appropriate token PIN.

This is the same as the request in #37 where I provided some sample code. Although that is marked resolved, v0.5.1 does not send the transactionID with the response during an OTP challenge (e.g. SMS) - the user must re-enter the token PIN as well as the OTP.

(Not clear if this fix was lost in the migration to java-sdk - I didn't test v0.5 or lower).

I found that adding the transactionID to the validateCheck call appears sufficient - tokens are now working as expected for each of the types, including when a user has multiple tokens and selects between them with different PINs.

I can create a PR if you think this is the appropriate fix:

```java
--- a/src/main/java/org/privacyidea/authenticator/PrivacyIDEAAuthenticator.java
+++ b/src/main/java/org/privacyidea/authenticator/PrivacyIDEAAuthenticator.java
@@ -198,7 +198,7 @@ public class PrivacyIDEAAuthenticator implements org.keycloak.authentication.Aut
             }
         } else {
             String otp = formData.getFirst(Const.FORM_PI_OTP);
-            PIResponse response = privacyIDEA.validateCheck(currentUserName, otp);
+            PIResponse response = privacyIDEA.validateCheck(currentUserName, otp, transactionID);
 
             if (response != null) {
                 // A challenge was triggered, display its message and pass the transaction id

```

### PR DESCRIPTION
**Pull request by nilsbehlen**
Closes #53 
Closes #58

### CODE REVIEW

**Review by plettich — COMMENTED** [review]
Looks good, just some comments/questions

File: src/main/java/org/privacyidea/authenticator/PrivacyIDEAAuthenticatorFactory.java [hunk 1]
```diff
@@ -76,84 +76,107 @@ public boolean isConfigurable() {
...
+        piRealm.setHelpText("Select the realm where your users are stored. Leave empty to use the default realm which is configured in the privacyIDEA server.");
         configProperties.add(piRealm);
 
         ProviderConfigProperty piVerifySSL = new ProviderConfigProperty();
         piVerifySSL.setType(ProviderConfigProperty.BOOLEAN_TYPE);
-        piVerifySSL.setName(Const.CONFIG_VERIFYSSL);
+        piVerifySSL.setName(Const.CONFIG_VERIFY_SSL);
         piVerifySSL.setLabel("Verify SSL");
-        piVerifySSL.setHelpText("Do not uncheck this in productive environment");
+        piVerifySSL.setHelpText("Do not set this to false in a productive environment. Disables the verification of the privacyIDEA server's certificate and hostname.");
```
[the remark this answers is not in the document]
Comment (nilsbehlen) [reply src/main/java/org/privacyidea/authenticator/PrivacyIDEAAuthenticatorFactory.java:94]: isn't it the possessive form?

File: src/main/java/org/privacyidea/authenticator/PrivacyIDEAAuthenticatorFactory.java [hunk 2]
```diff
@@ -76,84 +76,107 @@ public boolean isConfigurable() {
...
+        piServerUrl.setLabel("privacyIDEA URL");
         piServerUrl.setHelpText("The URL of the privacyIDEA server (complete with scheme, host and port like \"https://<piserver>:port\")");
         configProperties.add(piServerUrl);
 
         ProviderConfigProperty piRealm = new ProviderConfigProperty();
         piRealm.setType(ProviderConfigProperty.STRING_TYPE);
         piRealm.setName(Const.CONFIG_REALM);
         piRealm.setLabel("Realm");
-        piRealm.setHelpText("Select the realm where your users are stored. Leave empty for default.");
+        piRealm.setHelpText("Select the realm where your users are stored. Leave empty to use the default realm which is configured in the privacyIDEA server.");
```
[the remark this answers is not in the document]
Comment (nilsbehlen) [reply src/main/java/org/privacyidea/authenticator/PrivacyIDEAAuthenticatorFactory.java:87]: I don't really know if we can get the (keycloak) locale in that class, might look into that someday.

File: src/main/resources/theme-resources/resources/pi-webauthn.js [hunk 3]
```diff
@@ -0,0 +1,514 @@
+/*
```
[the remark this answers is not in the document]
Comment (nilsbehlen) [reply src/main/resources/theme-resources/resources/pi-webauthn.js:1]: If i understand it correctly, the js is complete and we will not need to touch it in the future just like the u2f js which deployed with the plugins. Will look into this anyway.

File: src/main/java/org/privacyidea/authenticator/PrivacyIDEAAuthenticatorFactory.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (plettich) [reply src/main/java/org/privacyidea/authenticator/PrivacyIDEAAuthenticatorFactory.java:94]: You are right.

File: src/main/resources/theme-resources/resources/pi-webauthn.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (plettich) [reply src/main/resources/theme-resources/resources/pi-webauthn.js:1]: We should look into this for the multiple token issue. I'll accept this for now.

**Pull request merged**